src/water_cmd.cpp

```diff
@@ -403,6 +410,7 @@ static CommandCost RemoveLock(TileIndex tile, DoCommandFlag flags)
 			MakeRiver(tile, Random());
 		} else {
 			DoClearSquare(tile);
+			ClearNeighbourNonFloodingStates(tile);
```

Comment:
This line is not required. `MakeWater` will be called which resets the state itself.

Reply:
MakeWater is not called for `tile`, only `tile +/- delta`

Reply:
I meant to say `MakeWaterKeepingClass` below will call `MakeWater` on the `tile +/- delta`.